src/TableNew/index.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTable } from 'react-table';
+import Table from './Table';
+
+import TablePagination from './TablePagination';
+import getVisibleColumns from './utils/getVisibleColumns';
+import { requiredWhen } from './utils/propTypesUtils';
+import getTableArgs from './utils/getTableArgs';
+import TableControlBar from './TableControlBar';
+
+function TableWrapper({
+  columns, data, bulkActions, defaultColumnValues, additionalColumns, isSelectable, isSortable,
+  isPaginated, manualPagination, pageCount, itemCount,
+  isFilterable, manualFilters, fetchData, initialState,
+}) {
+  const defaultColumn = React.useMemo(
+    () => (defaultColumnValues),
+    [defaultColumnValues],
+  );
+  const tableOptions = {
+    columns,
+    data,
+    defaultColumn,
+    manualFilters,
+    manualPagination,
+    initialState,
+  };
+
+  if (isPaginated) {
+    tableOptions.pageCount = pageCount || itemCount % initialState.pageSize || -1;
+  }
+
+  // NB: Table args *must* be in a particular order
+  const tableArgs = getTableArgs({
+    tableOptions, isFilterable, isSelectable, isPaginated, isSortable,
+  });
+  // adds selection column and action columns as necessary
+  tableArgs.push(hooks => {
+    hooks.visibleColumns.push(visibleColumns => getVisibleColumns(isSelectable, visibleColumns, additionalColumns));
+  });
+
+  // Use the state and functions returned from useTable to build your UI
+  const instance = useTable(...tableArgs);
+
+  useEffect(() => {
+    if (fetchData) {
+      const currentState = {};
+      if (manualFilters) {
+        currentState.currentFilters = instance.state.filters;
+      }
+      if (manualPagination) {
+        currentState.pageSize = instance.state.pageSize;
+        currentState.pageIndex = instance.state.pageIndex;
+      }
+      fetchData(currentState);
+    }
+  }, [instance.state.pageSize, instance.state.filters, instance.state.pageIndex, fetchData]);
```

Comment:
Should / does `fetchData` need to be called when sorting by a column? At first glance in the code and playing around the Paragon docs site example of backend filtering, it doesn't appear that it will re-fetch data for column sorting.

Reply:
We're not currently supporting backend sorting. Unless the table is _huge_ and not paginated, react-table sorts quickly enough for our needs. It's an easily addable feature, I can put it on the v1.1 list!

Reply:
If a table of any size is paginated and the user is on page 1 when they sort a column, how will the table know what data to sort from page 2 and beyond if it doesn't support backend sorting?

Reply:
Sorting feature has now been added.